### PR TITLE
ports the memory inhibitor from aurora station, and renames it to "class H amnestic implant"

### PR DESCRIPTION
**Pull request by quadbium**
## About the Pull Request
ported from: https://github.com/Aurorastation/Aurora.3/pull/11360

ports the memory inhibitor from aurora station, and renames it to "class H amnestic implant"
adds the class H amnestic implant as an augment that can be taken at round start.
## Why It's Good For The Game

adds RP options for secretary and admin assistant players, who can now be more easily amnesticized after bearing witness to high security interactions.

## Changelog

:cl:
add: Added class H amnestic implant
add: Added the option to take the class H amnestic implant in the loadout
add: Changed the activation deactivation alerts to be in line with class H amnestics
/:cl:

### CODE REVIEW

**Comment by tichys** [comment]
I wish we could port everything from aurora

**Comment by Tsurupeta** [comment]
What's the point of it besides some flavor text?

**Comment by aidan-bowman** [comment]
> What's the point of it besides some flavor text?

You could say the same thing about Amnestics in general.

File: code/modules/augment/active/memory_inhibitor.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		ready_to_erase = TRUE
+		M.visible_message(SPAN_WARNING("[M]'s eyes grow dim."))
+		to_chat(M, "<font size='5' color='red'>It feels like a haze falls in your head... You can remember everything just fine, but you'll forget what happens later on.</font>")
+		ready_to_erase = TRUE
+	else
+		ready_to_erase = FALSE
+		M.visible_message(SPAN_WARNING("[M]'s eyes regain their focus."))
+		to_chat(M, "<font size='5' color='red'>Your mind feels a lot clearer, but... You can't recall the events since the last time you activated your memory inhibitor!</font>")
+
+/obj/item/organ/internal/augment/active/memory_inhibitor/emp_act(mob/living/M, removed)
```
Comment (Tsurupeta) [inline code/modules/augment/active/memory_inhibitor.dm]: ???

File: code/modules/augment/active/memory_inhibitor.dm [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	if(owner)
+		M.visible_message(SPAN_WARNING("[M] looks confused."))
+		to_chat(M, "<font size='5' color='red'>You forgot everything that happened today!</font>")
+	return TRUE
+
+/obj/item/organ/internal/augment/active/memory_inhibitor/emp_act(severity)
```
Comment (Tsurupeta) [inline code/modules/augment/active/memory_inhibitor.dm:24]: When it receives emp it calls emp again???

Comment (quadbium) [reply code/modules/augment/active/memory_inhibitor.dm:24]: fixed EMP act, no idea how to fix visible message.

File: code/modules/augment/active/memory_inhibitor.dm [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		ready_to_erase = TRUE
+	else
+		ready_to_erase = FALSE
+		M.visible_message(SPAN_WARNING("[M]'s eyes regain their focus."))
+		to_chat(M, "<font size='5' color='red'>Your mind feels a lot clearer, but... You can't recall the events since the last time you activated your memory inhibitor!</font>")
+
+/obj/item/organ/internal/augment/active/memory_inhibitor/emp_act(severity)
+	. = ..()
+	if(severity)
+		//M.visible_message(SPAN_WARNING("[M] looks confused"))
```
Comment (Tsurupeta) [inline code/modules/augment/active/memory_inhibitor.dm]: ?????????

Comment (Tsurupeta) [reply code/modules/augment/active/memory_inhibitor.dm]: Whats even broken with it and why is it commented out then if you cant "fix" it?

Comment (quadbium) [reply code/modules/augment/active/memory_inhibitor.dm]: figured out a way to do the "M" variable

File: code/modules/augment/active/memory_inhibitor.dm [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	else
+		ready_to_erase = FALSE
+		M.visible_message(SPAN_WARNING("[M]'s eyes regain their focus."))
+		to_chat(M, "<font size='5' color='red'>Your mind feels a lot clearer, but... You can't recall the events since the last time you activated your memory inhibitor!</font>")
+
+/obj/item/organ/internal/augment/active/memory_inhibitor/emp_act(severity)
+	if (istype(src.loc, /mob/living/carbon/human))
+		var/mob/living/carbon/human/M = src.loc
+		M.visible_message(SPAN_WARNING("[M] looks confused"))
+		to_chat(M, "<font size='5' color='red'>You forgot everything that happened today!</font>")
```
Comment (Tsurupeta) [inline code/modules/augment/active/memory_inhibitor.dm]: All the M's should be switched to owner var

**Review by Tsurupeta — CHANGES_REQUESTED** [review]

**Comment by quadbium** [comment]
@Tsurupeta turns out using the proper variable made everything much easier.

File: code/modules/augment/active/memory_inhibitor.dm [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/obj/item/organ/internal/augment/active/memory_inhibitor
+	name = "class H amnestic implant"
+	allowed_organs = list(BP_AUGMENT_HEAD)
+	icon_state = "memory_inhibitor"
+	desc = "An implant that allows one to have control over their memories, allowing you to set a timer and remove any memories developed within it. This is most popular with the SCP foundation."
+	action_button_name = "Activate Class H Amnestic Implant"
+	origin_tech = list(TECH_MATERIAL = 5, TECH_BIO = 5, TECH_ESOTERIC = 5)
+	var/ready_to_erase = FALSE
+
+/obj/item/organ/internal/augment/active/memory_inhibitor/attack_self(mob/living/owner, removed)
```
Comment (Tsurupeta) [inline code/modules/augment/active/memory_inhibitor.dm]:
```suggestion
/obj/item/organ/internal/augment/active/memory_inhibitor/activate()
```

File: code/modules/augment/active/memory_inhibitor.dm [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	. = ..()
+
+	if(!.)
+		return FALSE
```
Comment (Tsurupeta) [inline code/modules/augment/active/memory_inhibitor.dm]:
```suggestion
	if(!can_activate())
		return
```

**Review by Tsurupeta — APPROVED** [review]
ok whatever

**Pull request merged**